yt_dlp/extractor/adn.py

```diff
@@ -255,3 +280,44 @@ def _real_extract(self, url):
             'average_rating': float_or_none(video.get('rating') or metas.get('rating')),
             'comment_count': int_or_none(video.get('commentsCount')),
         }
+
+
+class ADNSeasonIE(ADNIE):
```

Comment:
Subclassing from another concrete extractor invites so many unexpected bugs. Create a `ADNBaseIE` with common methods from which both can be derived